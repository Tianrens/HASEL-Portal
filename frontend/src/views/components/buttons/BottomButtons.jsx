import { React, useState } from 'react';
import Icon from '@mui/material/Icon';
import { StyledButton } from './StyledButton';
import { buttonContainer } from './BottomButtons.module.scss';
import ConfirmDialog from '../Modal/ConfirmDialog';

const BottomButtons = ({
    onDelete,
    deleteText,
    onDeny,
    denyText,
    onAccept,
    acceptText,
    acceptDisabled,
    deleteMessage,
    denyMessage,
    acceptMessage,
}) => {
    const [open, setOpen] = useState(false);
    const [message, setMessage] = useState('');
    const [onSubmit, setOnSubmit] = useState();

    const openModal = (msg, fn) => {
        setMessage(msg);
        setOnSubmit(() => fn);
        setOpen(true);
    };

    return (
        <>
            <div className={buttonContainer}>
                <div>
                    {onDelete && (
                        <StyledButton
                            color='red'
                            icon={<Icon>delete</Icon>}
                            type='submit'
                            onClick={() => openModal(deleteMessage, onDelete)}
                        >
                            {deleteText ?? 'Delete'}
                        </StyledButton>
                    )}
                </div>
                <div>
                    {onDeny && (
                        <StyledButton
                            color="lighter"
                            outline
                            icon={<Icon>close</Icon>}
                            type='submit'
                            onClick={() => openModal(denyMessage, onDeny)}
                        >
                            {denyText ?? 'Deny'}
                        </StyledButton>
                    )}
                    {onAccept && (
                        <StyledButton
                            icon={<Icon>done</Icon>}
                            type='submit'
                            onClick={() => openModal(acceptMessage, onAccept)}
                            disabled={acceptDisabled}
                        >
                            {acceptText ?? 'Accept'}
                        </StyledButton>
                    )}
                </div>
            </div>
            <ConfirmDialog
                message={message}
                open={open}
                onSubmit={onSubmit}
                handleClose={() => setOpen(false)}
            />
        </>
    );
};

export default BottomButtons;
