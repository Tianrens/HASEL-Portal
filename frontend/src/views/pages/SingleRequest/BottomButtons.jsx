import { React } from 'react';
import Icon from '@material-ui/core/Icon';
import { StyledButton } from '../../components/buttons/StyledButton';
import styles from './SingleRequest.module.scss';

const BottomButtons = ({ onDelete, onDeny, onAccept }) => (
    <div className={styles.buttonContainer}>
        <div>
            <StyledButton
                className={styles.button}
                color='yellow'
                icon={<Icon>delete</Icon>}
                type='submit'
                onClick={onDelete}
            >
                Delete
            </StyledButton>
        </div>
        <div>
            <StyledButton
                className={styles.button}
                color='red'
                icon={<Icon>close</Icon>}
                type='submit'
                onClick={onDeny}
            >
                Deny
            </StyledButton>
            <StyledButton
                className={styles.button}
                color='green'
                icon={<Icon>done</Icon>}
                type='submit'
                onClick={onAccept}
            >
                Accept
            </StyledButton>
        </div>
    </div>
);

export default BottomButtons;
