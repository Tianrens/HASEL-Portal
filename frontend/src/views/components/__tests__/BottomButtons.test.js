import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import BottomButtons from '../buttons/BottomButtons';

test('BottomButtons renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <BottomButtons onAccept={() => {}} onDeny={() => {}} onDelete={() => {}} />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
