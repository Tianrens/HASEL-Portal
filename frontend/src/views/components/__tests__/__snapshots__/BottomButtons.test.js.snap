// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomButtons renders properly 1`] = `
<div
  className="buttonContainer"
>
  <div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button yellow css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          delete
        </span>
      </span>
      Delete
    </button>
  </div>
  <div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button red css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          close
        </span>
      </span>
      Deny
    </button>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button green css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <span
          aria-hidden={true}
          className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
        >
          done
        </span>
      </span>
      Accept
    </button>
  </div>
</div>
`;
